src/persist-types/src/stats/primitive.rs

```diff
@@ -203,6 +198,20 @@ macro_rules! stats_primitive {
             fn none_count(&self) -> usize {
                 0
             }
+            fn downcast(stats: &super::ColumnarStats) -> Option<Self>
+            where
+                Self: Sized,
+            {
+                if stats.nulls.is_some() {
+                    return None;
+                }
+                match &stats.values {
```

Comment:
This PR has many copies of the last few lines, and it would be relatively easy to factor out a helper a la:

```rust
match stats.non_null_values()? {
```

OTOH, if you expect big changes or deletions for this stuff in the near future, not worth the trouble!